# Lab | Inferential statistics - ANOVA

### Part 1

In this activity, we will look at another example. Your task is to understand the problem and write down all the steps to set up ANOVA. After the next lesson, we will ask you to solve this problem using Python. Here are the steps that you would need to work on: - Null hypothesis - Alternate hypothesis - Level of significance - Test statistic - P-value - F table

### Context

Suppose you are working as an analyst in a microprocessor chip manufacturing plant. You have been given the task of analyzing a plasma etching process with respect to changing Power (in Watts) of the plasma beam. Data was collected and provided to you to conduct statistical analysis and check if changing the power of the plasma beam has any effect on the etching rate by the machine. You will conduct ANOVA and check if there is any difference in the mean etching rate for different levels of power. You can find the data anova_lab_data.xlsx file in the files_for_lab folder

State the null hypothesis

State the alternate hypothesis

What is the significance level
What are the degrees of freedom of model, error terms, and total DoF

Data was collected randomly and provided to you in the table as shown: link to the image - Data

### Null hypothesis
 Changing the power of the plasma beam does not have an effect on the etching rate by the machine

### Alternate hypothesis
Changing the power of the plasma beam has a significant effect on the etching rate

### Significance level
The significance level we will use is 0.05 or 5%

In [2]:
# Reading the file with the data
import pandas as pd

data = pd.read_excel("anova_lab_data.xlsx")

data

Power  Etching Rate
0   160 W          5.43
1   180 W          6.24
2   200 W          8.79
3   160 W          5.71
4   180 W          6.71
5   200 W          9.20
6   160 W          6.22
7   180 W          5.98
8   200 W          7.90
9   160 W          6.01
10  180 W          5.66
11  200 W          8.15
12  160 W          5.59
13  180 W          6.60
14  200 W          7.55

### Degrees of freedom of model, error terms, and total DoF

Degrees of freedom: number of power levels minus 1.. In our case the DoF are 2

Error terms: number of observations minus the number of power levels.. In this case the error terms are 12

Total DoF: number of observations minus 1.. For us is 14

### Part 2
In this section, use Python to conduct ANOVA.

What conclusions can you draw from the experiment and why?

In [3]:
# Importing necessary libraries
import pandas as pd
import scipy.stats as stats
import statsmodels.api as sm
from statsmodels.formula.api import ols

In [4]:
#changing the column names because I was having issues when fitting the model in next cell
data.columns = ["power", "etching_rate"]
print(data.columns)

Index(['power', 'etching_rate'], dtype='object')


In [4]:
# Fit the ANOVA model
model = ols("etching_rate ~ power", data=data).fit()

# Perform ANOVA
anova_table = sm.stats.anova_lm(model, typ=2)

In [5]:
# Print ANOVA
print(anova_table)

             sum_sq    df          F    PR(>F)
power     18.176653   2.0  36.878955  0.000008
Residual   2.957240  12.0        NaN       NaN


### Conclusion

The p-value for the power variable (0.000008) is significantly smaller than the usual threshold of 0.05. This indicates that changing the power of the plasma beam has a significant impact on the etching rate. 

Also the F-statistic value of 36.878955 shows us that there is a significant difference between the average etching rates of different power levels, the larger the F-statistic, the stronger is the evidence against the null hypothesis.

In summary, we can conclude that modifying the power of the plasma beam has a significant influence on the etching rate.